Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ShowViewshedFromPointOnMapView: View {
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    /// The point on the screen where the user tapped.
+    @State private var tapScreenPoint: Point?
+    /// The graphics overlay that shows where the user tapped on the screen.
+    @State private var inputGraphicsOverlay = {
+        let inputGraphicsOverlay = GraphicsOverlay()
+        // Red dot marker that is added to graphic overlay on map on the location where the user taps.
+        let pointSymbol = SimpleMarkerSymbol(style: .circle, color: .red, size: 10)
+        // Sets the renderer to draw the dot symbol on the graphics overlay.
+        let renderer = SimpleRenderer(symbol: pointSymbol)
+        inputGraphicsOverlay.renderer = renderer
+        return inputGraphicsOverlay
+    }()
+    /// The graphics overlay that displays the viewshed for the tapped location.
+    @State private var resultGraphicsOverlay = {
+        let resultGraphicsOverlay = GraphicsOverlay()
+        let fillColor = UIColor(
+            red: 226 / 255.0,
+            green: 119 / 255.0,
+            blue: 40 / 255,
+            alpha: 120 / 255.0
+        )
+        let fillSymbol = SimpleFillSymbol(
+            style: .solid,
+            color: fillColor,
+            outline: nil
+        )
+        let resultRenderer = SimpleRenderer(symbol: fillSymbol)
+        resultGraphicsOverlay.renderer = resultRenderer
+        return resultGraphicsOverlay
+    }()
+    /// The current draw status of the map.
+    @State private var currentDrawStatus: Bool = false
+    /// The processing task that references the online resource through the url.
+    @State private var geoprocessingTask: GeoprocessingTask = {
+        let geoprocess = GeoprocessingTask(url: .viewshedURL)
+        return geoprocess
+    }()
+    /// Handles the execution of the geoprocessing task and gets the result.
+    @State private var geoprocessingJob: GeoprocessingJob?
+    /// Sets map's initial viewpoint to Vanoise National Park in France.
+    @State private var map: Map = {
+        let map = Map(basemapStyle: .arcGISTopographicBase)
+        map.initialViewpoint = Viewpoint(
+            center: Point(latitude: 45.3790902612337, longitude: 6.84905317262762),
```

Comment:
I have to use this GIS cliché meme here again… https://xkcd.com/2170/ 😉 

```suggestion
            center: Point(latitude: 45.379, longitude: 6.849),
```